arrabiata/src/interpreter.rs

```diff
@@ -973,6 +1026,108 @@ pub fn run_ivc<E: InterpreterEnv>(env: &mut E, instr: Instruction) {
                 panic!("Invalid index: it is supposed to be less than {POSEIDON_ROUNDS_FULL}");
             }
         }
+        Instruction::PoseidonNextRow(curr_round) => {
+            env.activate_gadget(Gadget::PoseidonNextRow);
+            debug!("Executing instruction PoseidonNextRow({curr_round})");
+            if curr_round < POSEIDON_ROUNDS_FULL {
+                // Values to be absorbed are 0 when when the round is not zero,
+                // i.e. when we are processing the rounds.
+                let values_to_absorb: Vec<E::Variable> = (0..POSEIDON_STATE_SIZE - 1)
+                    .map(|_i| {
+                        let pos = env.allocate_public_input();
```

Comment:
I see that in this use case we absorb PI, but this is not always the case (eg. the internal node of a merkle tree)
Can we not harcode the fact that we absorb PI ?